File: cpp/Problem58.cpp
/*
 * Problem: 58. Length of Last Word
 * URL: https://leetcode.com/problems/length-of-last-word/
 * Difficulty: Easy
 * Tags: String
 */

#include <string>

using namespace std;

class Solution {
public:
	int lengthOfLastWord(string s) {
		int startIndex = -1, endIndex = -1;
		for (int i = s.length() - 1; i >= 0; i--) {
			if (s[i] != ' ') {
				endIndex = i;
				break;
			}
		}
		for (int i = 0; i <= endIndex; i++) {
			if (startIndex == -1 && s[i] != ' ') {
				startIndex = i;
			} else if (s[i] == ' ') {
				startIndex = -1;
			}
		}
		return endIndex - startIndex + 1;
	}
};

File: python/Problem58.py
# Problem: 58. Length of Last Word
# URL: https://leetcode.com/problems/length-of-last-word/
# Difficulty: Medium
# Tags: Easy


class Solution:
  def lengthOfLastWord(self, s: str) -> int:
    startIndex: int = -1
    endIndex: int = -1
    for i in range(len(s) - 1, -1, -1):
      if s[i] != ' ':
        endIndex = i
        break
    for i in range(0, endIndex + 1):
      if startIndex == -1 and s[i] != ' ':
        startIndex = i
      elif s[i] == ' ':
        startIndex = -1
    return endIndex - startIndex + 1